scripts/masker/dma-collection-masker

```diff
@@ -160,29 +315,34 @@ def _extract_collection(collection_file: Path, extract_path: Path) -> None:
         tf.close()
 
 
-def _find_file_referenced_by_rule(rule: DataMaskRule, search_path: Path) -> Optional[Path]:
-    """Find file referenced by data masking rule.
+def _find_files_referenced_by_rule(rule: DataMaskRule, search_path: Path) -> Optional[Path]:
+    """Find files referenced by data masking rule.
 
     Args:
         rule (DataMaskRule): The data masking rule to apply.
         search_path (Path): The search path to look for collection csv files.
 
     Raises:
-        ApplicationError: Raised when no matching file is found.
-        ApplicationError: Raise when more than 1 matching files are found.
+        ApplicationError: Raised when no matching files are found.
+        ApplicationError: Raise when more than 1 matching files are found if no wildcard search.
 
     Returns:
-       Optional[Path]: If the file was found, the file.  Else None.
+       Optional[list[Path]]: If the files were found, the files.  Else None.
     """
-    matched_file = list(search_path.glob(f"opdb__{rule.table_name}__*.csv"))
+    matched_file = list(
+        set(search_path.glob(f"opdb__{rule.table_name}__*.csv")).difference(search_path.glob("opdb__defines__*.csv"))
+    )
     if not matched_file and not rule.missing_ok:
-        raise ApplicationError(f"Could not find a file to match for {rule.table_name}")
+        msg = f"Could not find a file to match for {rule.table_name}"
+        raise ApplicationError(msg)
     if matched_file and not rule.missing_ok and len(matched_file) == 0:
-        raise ApplicationError(f"Could not find a file to match for {rule.table_name}")
-    if matched_file and len(matched_file) > 1:
-        raise ApplicationError(f"Found too many files when searching for {rule.table_name}.  Found {matched_file}")
-    if matched_file and len(matched_file) == 1:
-        return matched_file[0]
+        msg = f"Could not find a file to match for {rule.table_name}"
+        raise ApplicationError(msg)
+    if matched_file and len(matched_file) > 1 and rule.table_name != "*":
+        msg = f"Found too many files when searching for {rule.table_name}.  Found {matched_file}"
+        raise ApplicationError(msg)
+    if matched_file and len(matched_file) != 0:
+        return matched_file
```

Comment:
This now returns a list of paths rather than a single path as we now support * for table_list in a rule so this return is correct. In reviewing this comment I notice that the return typing of this function is incorrect and should be an optional list path.